setup.py

```diff
@@ -2,50 +2,40 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 import os
-import sys
 
+from setuptools import find_packages
 from setuptools import setup
-from setuptools.command.bdist_egg import bdist_egg
 
-# the name of the project
-name = 'jupyter_client'
-
-pjoin = os.path.join
 here = os.path.abspath(os.path.dirname(__file__))
-pkg_root = pjoin(here, name)
-
-packages = []
-for d, _, _ in os.walk(pjoin(here, name)):
-    if os.path.exists(pjoin(d, '__init__.py')):
-        packages.append(d[len(here) + 1 :].replace(os.path.sep, '.'))  # noqa
 
 version_ns = {}
-with open(pjoin(here, name, '_version.py')) as f:
+with open(os.path.join(here, 'jupyter_client', '_version.py')) as f:
     exec(f.read(), {}, version_ns)
 
+with open(os.path.join(here, 'README.md'), 'r') as f:
+    long_description = f.read()
 
-class bdist_egg_disabled(bdist_egg):
-    """Disabled version of bdist_egg
+with open(os.path.join(here, 'requirements.txt'), 'r') as f:
+    requirements = f.read().splitlines()
 
-    Prevents setup.py install from performing setuptools' default easy_install,
-    which it should never ever do.
-    """
+with open(os.path.join(here, 'requirements-test.txt'), 'r') as f:
+    requirements_test = f.read().splitlines()
 
-    def run(self):
-        sys.exit("Aborting implicit building of eggs. Use `pip install .` to install from source.")
+with open(os.path.join(here, 'requirements-doc.txt'), 'r') as f:
+    requirements_doc = f.read().splitlines()
 
-
-setup_args = dict(
-    name=name,
+setup(
+    name='jupyter_client',
     version=version_ns['__version__'],
-    packages=packages,
+    packages=find_packages(exclude=["docs", "docs.*", "tests", "tests.*"]),
     description='Jupyter protocol implementation and client libraries',
-    long_description=open('README.md').read(),
+    long_description=long_description,
     long_description_content_type='text/markdown',
     author='Jupyter Development Team',
     author_email='jupyter@googlegroups.com',
     url='https://jupyter.org',
     license='BSD',
+    license_file='LICENSE.md',
```

Comment:
`LICENSE.md` file is more common name which is used in other jupyter repos like `notebook`, `nbclient`, etc

Reply:
That might requires changes in the Conda recipe and other linux package build scripts. We should document it in change log.

Reply:
I can revert this change